src/platform/device.gni

```diff
@@ -89,6 +89,9 @@ declare_args() {
     chip_mdns = "none"
   }
 
+  # Enable factory data support
+  chip_enable_factory_data = false
```

Comment:
Note that it can be put in `platform/nrfconnect/args.gni` since this is nrfconnect-only.